scripts/linux/smokeTestHelper.sh

```diff
@@ -0,0 +1,220 @@
+#! /bin/bash
+
+get-latest-version-apt()
+{
+  # Use the 'apt' to get the latest version string from the linux repository
+  # $1 - package name
+  # $2 - the Nth latest package (ex. 1 is the latest, 2 is the second latest), Optional
+  
+  apt-cache madison $1 | awk -F '|' '{gsub(/ /,"",$2); print $2}' | sort --version-sort -r -u | sed -n ${2:-1}p
+}
+
+check-matching-version()
+{
+  # Compare if ($2) is a substring of ($1)
+  # $1 - version string 1 (string)
+  # $2 - version string 2 (string)
+  # $3 - pkgName
+  
+  if [[ ! "$1" =~ "$2" ]]; then
+    echo "[FAIL] Mismatch Version: The latest $pkgName package version on linux repo is $1 (expecting: $2)"
+    exit 1
+  fi
+  echo "[PASS] $pkgName linux repository artifact version check ($1)"
+  echo ""
+}
+
+check-file-size-limit()
+{
+  # Compare if the incoming file size is within $sizePercentThrehold percent of
+  # the golden package standard. That is 
+  #  goldenPkgSize*(100-sizePercentThrehold)/100 < incomingPkgSize < goldenPkgSize*(100+sizePercentThrehold)/100
+  #
+  # $1 - incoming artifact
+  # $2 - golden standard artifact
+  # $3 - acceptable size difference percentage as a whole integer number
+
+  incomingArtifact=$1
+  goldenArtifact=$2
```

Comment:
Should we also have 3rd arg?

Reply:
I used the `$3` directly without casting it into a local var.... I ran into an issue w/ Bash when it converts string into math mode -- using $3 seems to get around the problem.